pupil_src/shared_modules/calibration_routines/fingertip_calibration.py

```diff
@@ -0,0 +1,139 @@
+'''
+(*)~---------------------------------------------------------------------------
+Pupil - eye tracking platform
+Copyright (C) 2012-2018 Pupil Labs
+
+Distributed under the terms of the GNU
+Lesser General Public License (LGPL v3.0).
+See COPYING and COPYING.LESSER for license details.
+---------------------------------------------------------------------------~(*)
+'''
+
+from pyglui.cygl.utils import draw_points_norm, draw_polyline_norm, RGBA
+from pyglui.ui import Info_Text
+from OpenGL.GL import GL_POLYGON
+from glfw import GLFW_PRESS, GLFW_KEY_SPACE, GLFW_KEY_F, glfwSetKeyCallback
+import audio
+from fingertip_detector import FingertipTracker
+from . finish_calibration import finish_calibration
+from . calibration_plugin_base import Calibration_Plugin
+
+# logging
+import logging
+logger = logging.getLogger(__name__)
+
+
+class Fingertip_Calibration(Calibration_Plugin):
+    def __init__(self, g_pool):
+        super().__init__(g_pool)
+        self.pos = None
+        self.fingertip_tracker = FingertipTracker()
+        self.fingertip = None
+        self.press_key = 0
+
+        self.menu = None
+
+    def init_ui(self):
+        super().init_ui()
+        self.menu.label = 'Fingertip Calibration'
+        self.menu.append(Info_Text('Calibrate gaze parameters using finger'))
```

Comment:
also add:

self.menu.append(Info_Text('This is a experimental calibration routine. Move you finger in your field of view while looking at the finger tip. This plugin is for prototyping and experimentation only. The detection robustness is not production grade. We will put a lot more effort into this going forward but wanted to release the idea and hope for feedback!'))